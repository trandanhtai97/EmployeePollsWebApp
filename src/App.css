.container {
  width: 1200px;
  display: flex;
  flex-direction: column;
}
