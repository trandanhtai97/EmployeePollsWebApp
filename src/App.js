import "./App.css";
import { useEffect } from "react";
import { Routes, Route, Navigate } from "react-router-dom";

import Login from "pages/Login/Login";
import { useDispatch } from "react-redux";
import { fetchPolls } from "stores/reducers/polls";
import { fetchUsers } from "stores/reducers/users";
import AppRoutes from "components/AppRoutes/AppRoutes";
import Dashboard from "pages/Dashboard/Dashboard";
import NotFound from "pages/NotFound/NotFound";
import PollDetail from "pages/PollDetail/PollDetail";
import Leaderboard from "pages/Leaderboard/Leaderboard";
import CreatePoll from "pages/CreatePoll/CreatePoll";

const App = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchPolls());
    dispatch(fetchUsers());
  }, [dispatch]);

  return (
    <div className="container">
      <Routes>
        <Route path="/login" element={<Login />} />
        <Route element={<AppRoutes />}>
          <Route path="/" element={<Dashboard />} />
          <Route path="/question/:id" element={<PollDetail />} />
          <Route path="/leaderboard" element={<Leaderboard />} />
          <Route path="/new-poll" element={<CreatePoll />} />
        </Route>
        <Route path="/not-found" element={<NotFound />} />
        <Route path="*" element={<Navigate to="/not-found" />} />
      </Routes>
    </div>
  );
};

export default App;
