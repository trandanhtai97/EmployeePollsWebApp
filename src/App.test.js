import { render, screen } from "@testing-library/react";
import App from "App";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import mockStore from "tests/MockStore";
import testStore from "tests/TestStore";

let store;

beforeEach(() => {
  store = testStore(mockStore);
});

describe("App", () => {
  it("will render the application", () => {
    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    expect(screen.getByText("Home")).toBeInTheDocument();
  });
});
