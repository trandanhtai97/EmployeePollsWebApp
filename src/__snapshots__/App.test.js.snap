// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="nav-bar"
        >
          <div
            class="nav-bar-left"
          >
            <div
              class="nav-bar-item nav-bar-item-active"
              data-testid="home"
            >
              Home
            </div>
            <div
              class="nav-bar-item "
              data-testid="leaderboard"
            >
              Leaderboard
            </div>
            <div
              class="nav-bar-item "
              data-testid="newPoll"
            >
              New
            </div>
          </div>
          <div
            class="nav-bar-item-center"
          />
          <div
            class="nav-bar-right"
          >
            <div
              class="nav-bar-item nav-bar-item-user"
            >
              sarahedo
            </div>
            <div
              class="nav-bar-item "
              data-testid="logout"
            >
              Logout
            </div>
          </div>
        </div>
        <h1>
          Welcome 
          sarahedo
        </h1>
        <div
          class="nav-bar-left "
        >
          <div
            class="nav-bar-item nav-bar-item-active"
            data-testid="unansweredTab"
          >
            New Questions
          </div>
          <div
            class="nav-bar-item false"
            data-testid="answeredTab"
          >
            Done
          </div>
        </div>
        <div
          class="group-poll"
        >
          <div
            class="group-poll-left"
          />
          <div
            class="group-poll-center"
          >
            <div
              class="group-poll-label"
            >
              New Questions
            </div>
            <div
              class="polls"
            >
              <div
                class="poll"
              >
                <div
                  class="poll-item-top"
                >
                  <div
                    class="poll-item-top-user"
                  >
                    tylermcginnis
                  </div>
                  <div
                    class="poll-item-top-date"
                  >
                    2:12 AM | 9/17/2023
                  </div>
                </div>
                <div
                  class="poll-item-bottom"
                >
                  <a
                    class="poll-btn"
                    href="/question/vthrdm985a262al8qx3do"
                  >
                    Show
                  </a>
                </div>
              </div>
              <div
                class="poll"
              >
                <div
                  class="poll-item-top"
                >
                  <div
                    class="poll-item-top-user"
                  >
                    mtsamis
                  </div>
                  <div
                    class="poll-item-top-date"
                  >
                    11:32 PM | 9/15/2023
                  </div>
                </div>
                <div
                  class="poll-item-bottom"
                >
                  <a
                    class="poll-btn"
                    href="/question/xj352vofupe1dqz9emx13r"
                  >
                    Show
                  </a>
                </div>
              </div>
            </div>
          </div>
          <div
            class="group-poll-right"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="nav-bar"
      >
        <div
          class="nav-bar-left"
        >
          <div
            class="nav-bar-item nav-bar-item-active"
            data-testid="home"
          >
            Home
          </div>
          <div
            class="nav-bar-item "
            data-testid="leaderboard"
          >
            Leaderboard
          </div>
          <div
            class="nav-bar-item "
            data-testid="newPoll"
          >
            New
          </div>
        </div>
        <div
          class="nav-bar-item-center"
        />
        <div
          class="nav-bar-right"
        >
          <div
            class="nav-bar-item nav-bar-item-user"
          >
            sarahedo
          </div>
          <div
            class="nav-bar-item "
            data-testid="logout"
          >
            Logout
          </div>
        </div>
      </div>
      <h1>
        Welcome 
        sarahedo
      </h1>
      <div
        class="nav-bar-left "
      >
        <div
          class="nav-bar-item nav-bar-item-active"
          data-testid="unansweredTab"
        >
          New Questions
        </div>
        <div
          class="nav-bar-item false"
          data-testid="answeredTab"
        >
          Done
        </div>
      </div>
      <div
        class="group-poll"
      >
        <div
          class="group-poll-left"
        />
        <div
          class="group-poll-center"
        >
          <div
            class="group-poll-label"
          >
            New Questions
          </div>
          <div
            class="polls"
          >
            <div
              class="poll"
            >
              <div
                class="poll-item-top"
              >
                <div
                  class="poll-item-top-user"
                >
                  tylermcginnis
                </div>
                <div
                  class="poll-item-top-date"
                >
                  2:12 AM | 9/17/2023
                </div>
              </div>
              <div
                class="poll-item-bottom"
              >
                <a
                  class="poll-btn"
                  href="/question/vthrdm985a262al8qx3do"
                >
                  Show
                </a>
              </div>
            </div>
            <div
              class="poll"
            >
              <div
                class="poll-item-top"
              >
                <div
                  class="poll-item-top-user"
                >
                  mtsamis
                </div>
                <div
                  class="poll-item-top-date"
                >
                  11:32 PM | 9/15/2023
                </div>
              </div>
              <div
                class="poll-item-bottom"
              >
                <a
                  class="poll-btn"
                  href="/question/xj352vofupe1dqz9emx13r"
                >
                  Show
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          class="group-poll-right"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
